README.md

```diff
@@ -1458,6 +1459,177 @@ the properties section in the JSON configuration file. Lastly, the client calls
+    this.username = username;
+    this.password = Buffer.from(password, 'utf8');
+    this.endpoint = endpoint;
+}
+
+UsernamePasswordCredentials.prototype.readPortable = function (reader) {
+    this.username = reader.readUTF('principal');
+    this.endpoint = reader.readUTF('endpoint');
+    this.password = reader.readByteArray('pwd');
+};
+
+UsernamePasswordCredentials.prototype.writePortable = function (writer) {
+    writer.writeUTF('principal', this.username);
+    writer.writeUTF('endpoint', this.endpoint);
+    writer.writeByteArray('pwd', this.password);
+};
+
+UsernamePasswordCredentials.prototype.getFactoryId = function () {
+    return -1;
+};
+
+UsernamePasswordCredentials.prototype.getClassId = function () {
+    return 1;
+};
+
+exports.UsernamePasswordCredentials = UsernamePasswordCredentials;
+```
+
+And below is the `Factory` implementation for the `Portable` implementation of `UsernamePasswordCredentials`.
+
+**user_pass_cred_factory.js**
+```javascript
+var UsernamePasswordCredentials = require('./user_pass_cred').UsernamePasswordCredentials;
+
+function UsernamePasswordCredentialsFactory() {
+}
+
+UsernamePasswordCredentialsFactory.prototype.create = function (classId) {
+    if(classId === 1){
+        return new UsernamePasswordCredentials();
+    }
+    return null;
+};
+
+exports.UsernamePasswordCredentialsFactory = UsernamePasswordCredentialsFactory;
+```
+
+Now, you can start your client by registering the `Portable` factory and giving the credentials as follows.
+
+```javascript
+var Client = require('hazelcast-client').Client;
+var ClientConfig = require('hazelcast-client').Config.ClientConfig;
+
+var UsernamePasswordCredentials = require('./user_pass_cred').UsernamePasswordCredentials;
+var UsernamePasswordCredentialsFactory = require('./user_pass_cred_factory').UsernamePasswordCredentialsFactory;
+
+var config = new ClientConfig();
+config.serializationConfig.portableVersion = 1;
+config.serializationConfig.portableFactories[-1] = new UsernamePasswordCredentialsFactory();
+config.customCredentials = new UsernamePasswordCredentials('admin', 'password', '127.0.0.1');
+
+Client.newHazelcastClient(config).then(function (client) {
+    var map;
+    return client.getMap('importantMap').then(function (mp) {
+        map = mp;
+        return map.put('key', 'value');
+    }).then(function () {
+        return map.get('key');
+    }).then(function (value) {
+        console.log(value);
+        return client.shutdown();
+    });
+});
+```
+
+> NOTE: It is almost always a bad idea to write the credentials to wire in a clear-text format. Therefore, using TLS/SSL encryption is highly recommended while using custom credentials as described in [TLS/SSL section]((#61-tlsssl)).
+
+With Hazelcast's extensible, `JAAS` based security features you can do much more than just authentication. 
+See [JAAS code sample](code_samples/jaas_sample) to see how to perform access control checks on client operations based on user groups.
```

Comment:
See the [JAAS code sample](code_samples/jaas_sample) to learn how to .... on the client operations ...